reports-api/src/reports_api/resources/indigenous_nation.py

```diff
@@ -15,62 +15,100 @@
-
-
-@cors_preflight('GET')
-@API.route('/exists', methods=['GET', 'OPTIONS'])
+parser.add_argument(
+    "name",
+    type=str,
+    required=True,
+    help="Name of indigenous nation to be checked.",
+    location="args",
+    trim=True,
+)
+parser.add_argument(
+    "id",
+    type=int,
+    help="ID of the indigenous nation in case of updates.",
+    location="args",
+)
+
+
+@cors_preflight("GET")
+@API.route("/exists", methods=["GET", "OPTIONS"])
 class ValidateIndigenousNation(Resource):
     """Endpoint resource to check for existing indigenous nation."""
 
     @staticmethod
-    @cors.crossdomain(origin='*')
+    @cors.crossdomain(origin="*")
     @auth.require
     @API.expect(parser)
     @profiletime
     def get():
         """Check for existing indigenous nations."""
         args = parser.parse_args()
-        name = args['name']
-        instance_id = args['id']
-        return IndigenousNationService.check_existence(name=name, instance_id=instance_id), HTTPStatus.OK
-
-
-@cors_preflight('GET, DELETE, PUT')
-@API.route('/<int:indigenous_nation_id>', methods=['GET', 'PUT', 'DELETE', 'OPTIONS'])
+        name = args["name"]
+        instance_id = args["id"]
+        exists = IndigenousNationService.check_existence(
+            name=name, instance_id=instance_id
+        )
+        return (
+            {"exists": exists},
+            HTTPStatus.OK,
+        )
+
+
+@cors_preflight("GET, DELETE, PUT")
+@API.route("/<int:indigenous_nation_id>", methods=["GET", "PUT", "DELETE", "OPTIONS"])
 class IndigenousNation(Resource):
     """Endpoint resource to manage an indigenous nation."""
 
     @staticmethod
-    @cors.crossdomain(origin='*')
+    @cors.crossdomain(origin="*")
     @auth.require
     @profiletime
     def get(indigenous_nation_id):
         """Return details of an indigenous nation."""
-        return IndigenousNationService.find(indigenous_nation_id), HTTPStatus.OK
+        indigenous_nation = IndigenousNationService.find(indigenous_nation_id)
+        if indigenous_nation:
+            return indigenous_nation, HTTPStatus.OK
+        return (
+            f"Indigenous nation with id '{indigenous_nation_id}' not found.",
+            HTTPStatus.NOT_FOUND,
+        )
 
     @staticmethod
-    @cors.crossdomain(origin='*')
+    @cors.crossdomain(origin="*")
     @auth.require
     @profiletime
     def put(indigenous_nation_id):
         """Update and return an indigenous nation."""
-        indigenous_nation = IndigenousNationService.update_indigenous_nation(indigenous_nation_id, API.payload)
-        return indigenous_nation.as_dict(), HTTPStatus.OK
+        indigenous_nation_schema = IndigenousNationSchema()
+        try:
+            request_json = indigenous_nation_schema.load(API.payload)
+        except ValidationError as err:
+            return err.messages, HTTPStatus.BAD_REQUEST
+        exists = IndigenousNationService.check_existence(
```

Comment:
I would prefer to check this condition inside service and throw a custom exception called ResourceExistsError . Then catch it in the resource level and return Conflict http respose.

Reply:
Updated